packages/grafana-data/src/types/config.ts

```diff
@@ -65,6 +65,26 @@ export type PreloadPlugin = {
   version: string;
 };
 
+/** Supported OAuth services
+ *
+ * @public
+ */
+export type OAuth =
+  | 'github'
+  | 'gitlab'
+  | 'google'
+  | 'generic_oauth'
+  // | 'grafananet' Deprecated. Key always changed to "grafana_com"
+  | 'grafana_com'
+  | 'azuread'
+  | 'okta';
+
+/** Map of enabled OAuth services and their respective names
+ *
+ * @public
+ */
+export type OAuthSettings = Partial<Record<OAuth, { name: string; icon?: string }>>;
+
```

Comment:
```suggestion
/**
 * @public
 */
export type OAuthDescription = {
   name: string;
   icon?: string;
}

/**
 * @public
 */
export type OAuthSettings = {
   github?: OAuthDescription;
   gitlab?: OAuthDescription;
   google?: OAuthDescription;
   generic_oauth?: OAuthDescription;
   grafana_com?: OAuthDescription;
   azuread?: OAuthDescription;
   okta?: OAuthDescription;
};
```

Reply:
yeah, the original definition here wasn't made by me (I just added icon) and I'm backporting them here for completeness.  Could certainly be clearer that way if we wanted to change in main.